### PR TITLE
Remove hpx/runtime/threads_fwd.hpp and hpx/util_fwd.hpp

### PR DESCRIPTION
**Pull request by msimberg**
Removes two `_fwd.hpp` headers without compatibility options. Is someone actually relying on these? `hpx/lcos_fwd.hpp` is a bit trickier...

### CODE REVIEW

File: hpx/runtime/components/server/abstract_migration_support.hpp [hunk 1]
```diff
@@ -135,5 +135,4 @@ namespace hpx { namespace components
             return this->base_type::on_migrated();
         }
     };
-}}
-
+}}    // namespace hpx::components
```
Comment (hkaiser) [inline hpx/runtime/components/server/abstract_migration_support.hpp:138]: Should we format these files at this point?

File: hpx/runtime/components/server/migration_support.hpp [hunk 2]
```diff
@@ -239,5 +243,4 @@ namespace hpx { namespace components
         hpx::lcos::local::promise<void> trigger_migration_;
         bool was_marked_for_migration_;
     };
-}}
-
+}}    // namespace hpx::components
```
Comment (hkaiser) [inline hpx/runtime/components/server/migration_support.hpp:246]: Same here: should this be formatted now?

Comment (msimberg) [reply hpx/runtime/components/server/migration_support.hpp:246]: No, this was accidental. I'll stick to only formatting/sorting includes in these files.

File: hpx/runtime/components/server/abstract_migration_support.hpp (hunk 1 above)
Comment (msimberg) [reply hpx/runtime/components/server/abstract_migration_support.hpp]: Removed these changes.

File: hpx/runtime/components/server/migration_support.hpp (hunk 2 above)
Comment (msimberg) [reply hpx/runtime/components/server/migration_support.hpp]: Removed these too.

**Pull request merged**